src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/Criterion.java

```diff
@@ -1,14 +1,54 @@
 package io.github.oliviercailloux.y2018.apartments.valuefunction;
 
 public enum Criterion {
-  TELE,
-  TERRACE,
-  WIFI,
-  FLOOR_AREA,
-  FLOOR_AREA_TERRACE,
-  NB_BATHROOMS,
-  NB_BEDROOMS,
-  NB_SLEEPING,
-  NB_MIN_NIGHT,
-  PRICE_PER_NIGHT;
+  TELE(ValueFunctionType.IS_BOOLEAN_CRESCENT),
+  TERRACE(ValueFunctionType.IS_BOOLEAN_CRESCENT),
+  WIFI(ValueFunctionType.IS_BOOLEAN_CRESCENT),
+  FLOOR_AREA(ValueFunctionType.IS_NOT_BOOLEAN_CRESCENT),
+  FLOOR_AREA_TERRACE(ValueFunctionType.IS_NOT_BOOLEAN_CRESCENT),
+  NB_BATHROOMS(ValueFunctionType.IS_NOT_BOOLEAN_CRESCENT),
+  NB_BEDROOMS(ValueFunctionType.IS_NOT_BOOLEAN_CRESCENT),
+  NB_SLEEPING(ValueFunctionType.IS_NOT_BOOLEAN_CRESCENT),
+  NB_MIN_NIGHT(ValueFunctionType.IS_NOT_BOOLEAN_DECREASE),
+  PRICE_PER_NIGHT(ValueFunctionType.IS_NOT_BOOLEAN_DECREASE);
+
+  private ValueFunctionType valueFunctionType;
+
+  Criterion(ValueFunctionType valueFunctionType) {
+    this.valueFunctionType = valueFunctionType;
+  }
+
+  public ValueFunctionType getValueFunctionType() {
+    return this.valueFunctionType;
+  }
+
+  public boolean hasBooleanDomain() {
+    return this.valueFunctionType.equals(ValueFunctionType.IS_BOOLEAN_CRESCENT)
+        || this.valueFunctionType.equals(ValueFunctionType.IS_BOOLEAN_DECREASE);
+  }
+
+  public boolean hasDoubleDomain() {
+    return this.valueFunctionType.equals(ValueFunctionType.IS_NOT_BOOLEAN_CRESCENT)
+        || this.valueFunctionType.equals(ValueFunctionType.IS_NOT_BOOLEAN_DECREASE);
+  }
+
+  public boolean isValueFunctionType(ValueFunctionType vft) {
```

Comment:
Je ne suis pas sûre que cette fonction soit correctement nommée, peut-être plutôt hasValueFunctionTypeEqualsTo() (ou un autre nom, celui proposé est peut-être un peu long).

Reply:
Fix in d34a55ad54fb6e348608d341481c740cbb7ff702